Pacemaker/pcs_cluster_36_nodes_feasibility/scale-cluster-upto-36-nodes.md

```diff
@@ -0,0 +1,155 @@
+
+Below issue observed after scaling cluster and configuring resources
+- PCMK_ipc_buffer exceeded
+- Bad file descriptor related errors
+
+**Tunning performed**
+1. IPC buffer limit
+- Increased IPC buffer limit (PCMK_ipc_buffer=19725604)
+- IPC buffer is the size of message used to corosync for heartbeat communication.
+
+2. Open file limit
+- Increased open file limit to 1048575 (ulimit -n 1048575)
+- We observed bad file descriptor erros in the corosync logs and did above change to fix issue.
+
+3. Batch-limit
+- Batch-limit is the maximum number of actions that the cluster may execute in parallel across all nodes. The "correct" value will depend on the speed and load of your network and cluster nodes. If zero, the cluster will impose a dynamically calculated limit only when any node has high load.
+- We tried 25 fixed batch-limit instead of dynamic limit and observed that network increased upto 345 Mbits/s and due to high CPU usage DC node got also changes. We also observed that some resources got time-out during unstandby nodes operation.
+- We reverted batch-limit to zero and network load back to normal. So, We concluded batch-limit zero is suitable for large cluster.
+
+4. cluster-ipc-limit
+- CLuster-ipc-limit is the maximum IPC message backlog before one cluster daemon will disconnect another. This is of use in large clusters, for which a good value is the number of resources in the cluster multiplied by the number of nodes. The default of 500 is also the minimum. Raise this if you see "Evicting client" messages for cluster daemon PIDs in the logs.
+- We tried 10000 cluster-ipc-limit but not seen much improvement in the cluster.
+- As per corosync guide this cluster property is useful for large sized cluster and recommanded value is ( Number of Nodes * Resources count )
+
+## 4. Cluster status alternative
+- We observed that "pcs status" command is taking bit more time around 10 sec to 13 sec for the execution.
+- We figured out some other alternatives to get nodes and resources status which are much faster than "pcs status".
+
+```
+$ pcs status nodes # Returns the status of all the nodes including remote nodes which are present in the cluster. It took around 1-2 secs for the execution.
+$ pcs resource # Returns the resource status of the cluster. It took around 1-2 secs for the execution.
+```
+
+
+ ## 5. Nodes standby/Unstandby operation
+
+i. pcs cluster standby/unstandby --all
+- We observed pretty much high traffic max 90 Mbits/s on the network when we performed the standby/unstandby for all the nodes of the cluster.
+- Also observed that some services failed due to timeout.
+
+Solution:
+- We increased stop/start timeout from 20 sec to 50 sec but it did not help to fix the issue.
+- We divided all the nodes in the gorup of three and moved 3 nodes to standby/unstandby at a time once those are done with no active resources then move to the next 3 nodes. These solution went well and all the services stopped and started succesfully
```

Comment:
How much time entire cluster standby/unstandy takes?

Reply:
Actually, It depends on the number of configured resources. I measured the time when we perform standby/unstandby in group of three with 691 resources.
standby time - Approx. 8 mins
unstandby time - Approx. 19 mins